### DISCUSSION TITLE
Supporting/Updating language-shellscript

### DISCUSSION THREAD

**Discussion by stdedos** [github-linguist/linguist#6567] [Q&A] [answered]
Hello there! I'd like to provide examples for improving/extending ```` ```console````.

After some searching

https://github.com/github-linguist/linguist/blob/7ca3799b8b5f1acde1dd7a8dfb7ae849d3dfb4cd/lib/linguist/languages.yml#L6520-L6531

... and after a lot of searching, I found (at https://github.com/github-linguist/linguist/tree/master/vendor/grammars):

[language-shellscript @ 35dc1aa(Submodule)](https://github.com/atom/language-shellscript/tree/35dc1aa1371ab75ece08fd0837330eb62e1f9da9) | [Release v7.24.0 (#6162)](https://github.com/github-linguist/linguist/commit/59b2d88b2242e6062384e5fb876668cc30ead951) | last year
-- | -- | --

Which leads to the deprecated Atom repository 😕 

My examples are:

```console
u@h [2023-10-05 21:09:24]:~$ ssh -v host.com   # Fails?
u@h [2023-10-0520:38:08]:~$ ssh -v host.com    # _Should've worked, but ..._
ua@h [2023-10-0520:38:08]:~$ ssh -v host.com   # "It's missing just an `a`"?
ua@h [2023-10-05 21:09:24]:~$ ssh -v host.com  # _just for combinatorial completeness_
```

Where would it be appropriate to ask for this improvement?

**Comment by stdedos** [comment]
Also, _TIL_: `ShellSession:` fragments can be looked up at https://github.com/github-linguist/linguist/blob/master/vendor/README.md

**Reply by lildude** [reply, marked as answer]
Same place as usual… and [documented](https://github.com/github-linguist/linguist/blob/master/docs/troubleshooting.md#theres-a-problem-with-the-syntax-highlighting-of-a-file)… the upstream grammar, but as you've pointed out and discussed in https://github.com/github-linguist/linguist/issues/6293, the grammar is now deprecated because of Atom's deprecation so the only likely option is to find an alternate actively maintained grammar, submit your changes there and then open a PR here to switch grammars.
